src/gitingest/clone.py

```diff
@@ -86,15 +88,40 @@ async def clone_repo(config: CloneConfig, *, token: str | None = None) -> None:
 
     # Checkout the subpath if it is a partial clone
     if partial_clone:
-        subpath = config.subpath.lstrip("/")
-        if config.blob:
-            # When ingesting from a file url (blob/branch/path/file.txt), we need to remove the file name.
-            subpath = str(Path(subpath).parent.as_posix())
-
-        checkout_cmd = create_git_command(["git"], local_path, url, token)
-        await run_command(*checkout_cmd, "sparse-checkout", "set", subpath)
+        await _checkout_partial_clone(config, local_path, url, token)
 
     # Checkout the commit if it is provided
     if commit:
         checkout_cmd = create_git_command(["git"], local_path, url, token)
         await run_command(*checkout_cmd, "checkout", commit)
+
+
+def _checkout_partial_clone(config: CloneConfig, local_path: str, url: str, token: str | None) -> None:
+    """Handle sparse-checkout for partial clones.
+
+    This helper function sets the sparse-checkout configuration for a partial clone,
+    optionally adjusting the subpath if ingesting from a file URL.
+
+    Parameters
+    ----------
+    config : CloneConfig
+        The configuration for cloning the repository, including subpath and blob flag.
+    local_path : str
+        The local path where the repository has been cloned.
+    url : str
+        The URL of the repository.
+    token : str | None
+        GitHub personal access token (PAT) for accessing private repositories.
+        Can also be set via the ``GITHUB_TOKEN`` environment variable.
+
+    Returns
+    -------
+    None
+
+    """
+    subpath = config.subpath.lstrip("/")
+    if config.blob:
+        # When ingesting from a file url (blob/branch/path/file.txt), we need to remove the file name.
+        subpath = str(Path(subpath).parent.as_posix())
+    checkout_cmd = create_git_command(["git"], local_path, url, token)
+    return run_command(*checkout_cmd, "sparse-checkout", "set", subpath)
```

Comment:
We can just get the `local_path` and the `url` from the `config: CloneConfig`, right?

```suggestion
def _checkout_partial_clone(config: CloneConfig, token: str | None) -> None:
    """Handle sparse-checkout for partial clones.

    This helper function sets the sparse-checkout configuration for a partial clone,
    optionally adjusting the subpath if ingesting from a file URL.

    Parameters
    ----------
    config : CloneConfig
        The configuration for cloning the repository, including subpath and blob flag.
    token : str | None
        GitHub personal access token (PAT) for accessing private repositories.

    Returns
    -------
    None

    """
    subpath = config.subpath.lstrip("/")
    if config.blob:
        # When ingesting from a file url (blob/branch/path/file.txt), we need to remove the file name.
        subpath = str(Path(subpath).parent.as_posix())
    checkout_cmd = create_git_command(["git"], config.local_path, config .url, token)
    return run_command(*checkout_cmd, "sparse-checkout", "set", subpath)
```